### PR TITLE
Implementing session with block execution with ffi

### PR DESCRIPTION
**Pull request by amitsuryavanshi**
Allowing blocks to be passed to session with C implementation
`driver.session do |session|
    ...
 end`

`driver.session do
    ...
 end`

### CODE REVIEW

File: ffi/neo4j/driver.rb [hunk 1]
```diff
@@ -14,11 +14,15 @@
...
 require 'neo4j/driver/graph_database'
+require 'neo4j/driver/graph_database_ext'
 require 'neo4j/driver/internal_driver'
 require 'neo4j/driver/internal_record'
 require 'neo4j/driver/internal_session'
 require 'neo4j/driver/internal_statement_result'
 require 'neo4j/driver/types/node'
 require 'neo4j/driver/types/path'
 require 'neo4j/driver/types/relationship'
+require 'neo4j/driver/auto_closable'
```
Comment (klobuczek) [inline ffi/neo4j/driver.rb]: keep alphabetic order

File: ffi/neo4j/driver/graph_database_ext.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+module Neo4j
+  module Driver
+    module GraphDatabaseExt
```
Comment (klobuczek) [inline ffi/neo4j/driver/graph_database_ext.rb]: unnecessary module, see jruby dir for how the AutoClosable is added

File: ffi/neo4j/driver/internal_driver.rb [hunk 3]
```diff
@@ -16,9 +19,8 @@ def initialize(uri, auth_token)
         Bolt::Config.destroy(config)
       end
 
-      def session
+      def session(&block)
```
Comment (klobuczek) [inline ffi/neo4j/driver/internal_driver.rb]: unnecessary change. That's what autoclosable does

File: ffi/neo4j/driver/internal_driver.rb [hunk 4]
```diff
@@ -16,9 +19,8 @@ def initialize(uri, auth_token)
...
         status = Bolt::Status.create
-        # connection = Bolt::Connector.acquire(@connector, :bolt_access_mode_write, status)
```
Comment (klobuczek) [inline ffi/neo4j/driver/internal_driver.rb]: keep this comment, it is still work in progress

File: jruby/neo4j/driver.rb [hunk 5]
```diff
@@ -11,6 +10,7 @@
 require 'neo4j/driver/ext/ruby_converter'
 require 'neo4j/driver/ext/run_override'
 require 'neo4j/driver/version'
+require 'neo4j/driver/auto_closable'
```
Comment (klobuczek) [inline jruby/neo4j/driver.rb]: alphabetic order

File: lib/neo4j/driver/auto_closable.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Neo4j
+  module Driver
+    # prepending methods with block execution
```
Comment (klobuczek) [inline lib/neo4j/driver/auto_closable.rb]: Either document it properly or skip the comment. This does not help to understand. I plan to make it a separate gem where I would put more documentation, but it has to prove itself first here.

File: lib/neo4j_ruby_driver.rb [hunk 7]
```diff
@@ -6,4 +6,5 @@
 require 'neo4j/driver/exceptions/client_exception'
 require 'neo4j/driver/exceptions/no_such_record_exception'
 require 'neo4j/driver/exceptions/service_unavailable_exception'
+require 'neo4j/driver/auto_closable'
```
Comment (klobuczek) [inline lib/neo4j_ruby_driver.rb]: alphabetic order

File: ffi/neo4j/driver/internal_driver.rb (hunk 3 above)
Comment (amitsuryavanshi) [reply ffi/neo4j/driver/internal_driver.rb]: This is accidental.

File: ffi/neo4j/driver/internal_driver.rb [hunk 8]
```diff
@@ -18,7 +21,7 @@ def initialize(uri, auth_token)
 
       def session
         status = Bolt::Status.create
-        # connection = Bolt::Connector.acquire(@connector, :bolt_access_mode_write, status)
+         # connection = Bolt::Connector.acquire(@connector, :bolt_access_mode_write, status)
```
Comment (klobuczek) [inline ffi/neo4j/driver/internal_driver.rb]:
```suggestion
        # connection = Bolt::Connector.acquire(@connector, :bolt_access_mode_write, status)
```
